### DISCUSSION TITLE
Encoded URL not working in getStaticPaths

### DISCUSSION THREAD

**Discussion by saadq** [vercel/next.js#30007] [Help]
I have detailed most of the issue in my StackOverflow question: https://stackoverflow.com/questions/69609567/next-js-encodeuricomponent-doesnt-work-with-in-getstaticpaths. I'm going to be repeating most of the same details here.

---

 * Minimal repro site: https://nextjs-paths-issue.vercel.app/
 * Minimal repro code: https://github.com/saadq/nextjs-encoding-issue
   * [Home page](https://github.com/saadq/nextjs-encoding-issue/blob/main/src/pages/index.tsx) 
   * [Food page](https://github.com/saadq/nextjs-encoding-issue/blob/main/src/pages/food/%5BfoodTitle%5D.tsx)

I am trying to iterate through an array of food objects and create static pages for each one based on its title. This works for most of the foods, but if the food title contains a `/`, then navigating to the page (such as [the **"Nice strawberry/kiwis dessert"** page](https://nextjs-paths-issue.vercel.app/food/Nice%20strawberry%2Fkiwis%20dessert)) will throw a 404.

In the home page, I encode the URL when I [create the `Link`](https://github.com/saadq/nextjs-encoding-issue/blob/main/src/pages/index.tsx#L11) and then in the [`getStaticPaths` function](https://github.com/saadq/nextjs-encoding-issue/blob/main/src/pages/food/%5BfoodTitle%5D.tsx#L26), I create the `paths` using the same encoded link. However, it doesn't seem to work when deployed. 

The page does work locally when running `npm run dev`, but it seems that in the actual output build there are issues. Is there something I can do to allow paths with encoded slashes to work? Any help would be appreciated, thanks!

---

**Home page**

```tsx
const HomePage: NextPage = () => (
  <>
    <h1>Home</h1>
    <ul>
      {foods.map((food) => (
        <li key={food.title}>
          <Link href={`/food/${encodeURIComponent(food.title)}`}>
            {food.title}
          </Link>
        </li>
      ))}
    </ul>
  </>
)
```

**Food page**

```tsx
export const getStaticProps: GetStaticProps<Props, Params> = (ctx) => {
  const title = ctx.params?.foodTitle as string
  const food = foods.find((food) => food.title === title) as Food

  return {
    props: {
      food
    }
  }
}

export const getStaticPaths: GetStaticPaths = () => {
  const paths = foods.map((food) => `/food/${encodeURIComponent(food.title)}`)

  return {
    paths,
    fallback: false
  }
}

const FoodPage: NextPage<Props> = ({ food }) => {
  return (
    <>
      <Link href='/'>Go Back</Link>
      <h1>{food.title}</h1>
      <h2>Amount: {food.amount}</h2>
    </>
  )
}

export default FoodPage
```

**Comment by mayank1513** [comment]
> 

Facing same issue. Works well if I remove encodeURI
